src/utils/markets/getMarketsData.ts

```diff
@@ -0,0 +1,123 @@
+        aToken.decimals(overrides),
+        aToken.totalSupply(overrides),
+      ]);
+      const { variableDebtTokenAddress } = await lendingPool.getReserveData(underlying, overrides);
+      const debtToken = VariableDebtToken__factory.connect(variableDebtTokenAddress, provider);
+      const [
+        totalBorrow,
+        price,
+        { p2pIndexCursor },
+        { p2pBorrowAmount, poolBorrowAmount, p2pSupplyAmount, poolSupplyAmount },
+      ] = await Promise.all([
+        debtToken.totalSupply(overrides),
+        oracle.getAssetPrice(underlying, overrides),
+        lens.getMarketConfiguration(market, overrides),
+        lens.getMainMarketData(market, overrides),
+      ]);
+      const marketParameters: MarketMinimal = {
+        address: market,
+        totalSupply: totalSupply.mul(price).div(pow10BN(decimals)),
+        totalBorrow: totalBorrow.mul(price).div(pow10BN(decimals)),
+        totalMorphoSupply: p2pSupplyAmount.add(poolSupplyAmount),
+        totalMorphoBorrow: p2pBorrowAmount.add(poolBorrowAmount),
+        price,
+        p2pIndexCursor: BigNumber.from(p2pIndexCursor),
+      };
+      return marketParameters;
+    })
+  );
+};
+const getCompoundMarketsParameters = async (snapshotBlock: providers.BlockTag, provider: providers.Provider) => {
+  const morpho = MorphoCompound__factory.connect(addresses.morphoCompound.morpho, provider);
+  const overrides = { blockTag: snapshotBlock };
+  const allMarkets = await morpho
+    .getAllMarkets(overrides)
+    .then((r) => r.filter((market) => market.toLowerCase() !== cFei));
+  const comptroller = Comptroller__factory.connect(await morpho.comptroller(overrides), provider);
+  const oracle = CompoundOracle__factory.connect(await comptroller.oracle(overrides), provider);
+  const lens = MorphoCompoundLens__factory.connect(addresses.morphoCompound.lens, provider);
+
+  return await Promise.all(
+    allMarkets.map(async (market) => {
+      market = market.toLowerCase();
+      const cToken = CToken__factory.connect(market, provider);
+      const [
+        totalSupplyRaw,
+        supplyIndex,
+        totalBorrow,
+        price,
+        { p2pIndexCursor },
+        { p2pBorrowAmount, poolBorrowAmount, p2pSupplyAmount, poolSupplyAmount },
+      ] = await Promise.all([
+        cToken.totalSupply(overrides),
+        cToken.exchangeRateStored(overrides),
+        cToken.totalBorrows(overrides),
+        oracle.getUnderlyingPrice(market, overrides),
+        morpho.marketParameters(market, overrides),
+        lens.getMainMarketData(market, overrides),
+      ]);
+      const totalSupply = totalSupplyRaw.mul(supplyIndex).div(WAD);
+      const marketParameters: MarketMinimal = {
+        address: market,
+        totalSupply: totalSupply.mul(price).div(WAD),
+        totalBorrow: totalBorrow.mul(price).div(WAD),
+        price,
+        totalMorphoSupply: p2pSupplyAmount.add(poolSupplyAmount),
+        totalMorphoBorrow: p2pBorrowAmount.add(poolBorrowAmount),
```

Comment:
same question

Reply:
$ conversion introduce come computation approximation that are not necessary here